src/commands.ts

```diff
@@ -91,6 +91,30 @@ export function registerCommands(context: vscode.ExtensionContext, prManager: IP
 		});
 	}));
 
+	context.subscriptions.push(vscode.commands.registerCommand('pr.merge', async (pr?: PRNode, message?: string) => {
+		const pullRequest = ensurePR(prManager, pr);
+		return vscode.window.showWarningMessage(`Are you sure you want to merge this pull request on GitHub?`, 'Yes', 'No').then(async value => {
+			if (value === 'Yes') {
+				try {
+					let newComment: Comment;
+					if (message) {
+						newComment = await prManager.createIssueComment(pullRequest, message);
+					}
+
+					let newPR = await prManager.mergePullRequest(pullRequest);
+					vscode.commands.executeCommand('pr.refreshList');
+					_onDidUpdatePR.fire(newPR);
+					return newComment;
```

Comment:
Instead of creating and returning a new comment, let's return the result of `mergePullRequest`. Peng pointed out that we don't want to create comments if the action fails, so I think `pr.close` is also wrong, but we can fix it separately.

If it failed, we could display the message with `vscode.window.showErrorMessage`

Reply:
Fixed only `pr.merge`.
However, `pr.close` still return newComment.